src/software/gui/robot_diagnostics/widgets/feedback.cpp

```diff
@@ -1,3 +1,27 @@
 #include "software/gui/robot_diagnostics/widgets/feedback.h"
 
-void setupFeedback(Ui::AutoGeneratedMainWidget *widget) {}
+void setupFeedback(Ui::AutoGeneratedMainWidget* widget) {}
+
+void updateFeedback(Ui::AutoGeneratedMainWidget* widget, const SensorMsg& sensor_msg)
+{
+    for (const auto& robot_msg : sensor_msg.tbots_robot_msgs())
+    {
+        // update status progress bars
+        widget->progressBar_battery->setValue(
+            static_cast<int>(robot_msg.power_status().battery_voltage()));
```

Comment:
Didn't realize that progress bars use ints. Can we give the progress bars 2 decimal places of accuracy by multiplying and dividing by 100? We should make our own progress bar class like what we do for sliders (slider.h/cpp) to manage all that and it should do `QProgressBar::setFormat`to make it display the value properly

Reply:
added